stl/inc/expected

```diff
@@ -1159,21 +1159,21 @@ public:
 
     template <class _Uty>
     _NODISCARD friend constexpr bool operator==(const expected& _Left, const _Uty& _Right)
-        noexcept(noexcept(static_cast<bool>(_Left._Value == _Right))) /* strengthened */ {
+        noexcept(noexcept(_Left._Value == _Right)) /* strengthened */ {

```

Comment:
I think we should also detect the noexcept-ness of implicit conversion to `bool`. It's possible to create a contrived result type such that `noexcept(_Left._Value == _Right)` is `true` but implicitly converting the result to `bool` calls a potentionally-throwing conversion function.
```suggestion
        noexcept(noexcept(_STD _Fake_copy_init<bool>(_Left._Value == _Right))) /* strengthened */ {
```
(ditto below)

Reply:
An implementer and a user had to cross a river. The user said, "take me with you! I won't sting you with my poisonous fake boolean!". The implementer was required by the Standard to carry the user, who inevitably tried to sting the implementer halfway across the river. But the implementer's skin was titanium alloy, and they made it safely across.